### DISCUSSION TITLE
xrdp session dont start child sessions of some applications

### DISCUSSION THREAD

**Discussion by axelmaning00** [neutrinolabs/xrdp#2564] [Q&A]
Hi! I connect to Arch Linux remotely with xrdp. On my remote arch linux, I run various remote work sessions to further connect to other machines. That is, the connection diagram looks something like this
mypc (win10mstsc) => archlinux (with xrdp) => other machine, to connect to other machines I use different clients (uvnc, rdp, anydesk) and if rdp and vnc work well as plugins in remmina, then you need to use anydesk separately . There is a problem. ANYDESK does not work in an xrdp session. If you run Anydesk locally, then everything works.
P.S. sorry my very bad eng
How to make anydesk work under xrdp normally? I heard that there is some problem with PAM from which pulseaudio, for example, did not boot, maybe this is somehow connected?
Forgive me for writing this thread, I'm new to linux and I'm not able to understand all this on my own, I usually read forums to solve problems, but I want this particular topic there are only a couple of branches on the Internet, and all of them are without full solutions .

Also every time to restart Arch, you need to manually enter sudo reboot in the console, since rebooting according to the description from KDE Plasma does not work.

**Comment by matt335672** [comment]
Hi @axelmaning00 

Welcome to the world of Linux! Also, don't apologise for your English. The only way to learn a language is to use it, and you seem to be doing OK to me.

We need more information on what you mean by "ANYDESK does not work in an xrdp session"? 

It looks like anydesk allows you to generate [trace files](https://support.anydesk.com/knowledge/trace-files). Could you generate one for a working session (locally) and also one for an xrdp session and post them here?

Also, what is the output of the command `env | sort` on both a working and xrdp session?

The `reboot` behaviour is normal. See #2305 for a discussion on this.

**Reply by axelmaning00** [reply]
This local session, anydesk working
[env work.txt](https://github.com/axelmaning00/anydesk.xrdp/blob/main/env%20work.txt)
[anydesk.trace](https://github.com/axelmaning00/anydesk.xrdp/blob/main/anydeskwork.trace)
This xrdp session:
[env dont work xrdp](https://github.com/axelmaning00/anydesk.xrdp/blob/main/env%20not%20work.txt)
anydesk.trace file not created (anydesk not working)

**Comment by matt335672** [comment]
This won't be good (from the environment):-

```
HOME=/root
. . .
USER=root
```

This could well be #1684. Are you getting messages like this in `/var/log/xrdp-sesman.log`:-

```
pam_systemd(xrdp-sesman:session): Failed to create session: No child processes
```

If you are, the two workarounds for this problem are:-

1) If you're not using systemd to mount home directories, comment-out all calls to `pam_systemd_home.so` in `/etc/pam.d/system-auth`. See [this post](https://github.com/neutrinolabs/xrdp/issues/1684#issuecomment-717179597) and [this post](https://bbs.archlinux.org/viewtopic.php?id=258716)
2) Build and install the pam_close_systemd_system_dbus module. See [this post](https://github.com/neutrinolabs/xrdp/issues/1684#issuecomment-717904769)
